quickstart/101-azfw-with-fwpolicy/main.tf

```diff
@@ -0,0 +1,153 @@
+    azurerm_resource_group.azfw_rg,
+    azurerm_virtual_network.azfw_vnet
+  ]
+}
+
+// Create a Public IP Address for Azure Firewall
+resource "azurerm_public_ip" "pip_azfw" {
+  name                = "pip-azfw"
+  location            = azurerm_resource_group.azfw_rg.location
+  resource_group_name = azurerm_resource_group.azfw_rg.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  tags                = azurerm_resource_group.azfw_rg.tags
+  depends_on = [
+    azurerm_resource_group.azfw_rg
+  ]
+}
+
+// Create a Azure Firewall Policy
+resource "azurerm_firewall_policy" "azfw_policy" {
+  name                     = "azfw-policy"
+  resource_group_name      = azurerm_resource_group.azfw_rg.name
+  location                 = azurerm_resource_group.azfw_rg.location
+  sku                      = var.fw_sku
+  threat_intelligence_mode = "Alert"
+}
+
+// Create a Network Rule Collection Group
+// Create a Network Rule Collection
+// Create rules for NTP
+resource "azurerm_firewall_policy_rule_collection_group" "net_policy_rule_collection_group" {
+  name               = "DefaultNetworkRuleCollectionGroup"
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  priority           = 200
+  network_rule_collection {
+    name     = "DefaultNetworkRuleCollection"
+    action   = "Allow"
+    priority = 200
+    rule {
+      name                  = "time-windows"
+      protocols             = ["UDP"]
+      source_ip_groups      = [azurerm_ip_group.workload_ip_group.id, azurerm_ip_group.infra_ip_group.id]
+      destination_ports     = ["123"]
+      destination_addresses = ["132.86.101.172"]
+    }
+  }
+}
+
+// Create a Azure Firewall Policy Rule Collection Group
+// Create a Application Rule Collection
+// Create rules for Windows Update
+// Create rules for Microsoft.com
+resource "azurerm_firewall_policy_rule_collection_group" "app_policy_rule_collection_group" {
+  name               = "DefaulApplicationtRuleCollectionGroup"
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  priority           = 300
+  application_rule_collection {
+    name     = "DefaultApplicationRuleCollection"
+    action   = "Allow"
+    priority = 500
+    rule {
+      name = "AllowWindowsUpdate"
+
+      description = "Allow Windows Update"
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      source_ip_groups      = [azurerm_ip_group.workload_ip_group.id, azurerm_ip_group.infra_ip_group.id]
+      destination_fqdn_tags = ["WindowsUpdate"]
+    }
+    rule {
+      name        = "Global Rule"
+      description = "Allow access to Microsoft.com"
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      destination_fqdns = ["*.microsoft.com"]
+      terminate_tls     = false
+      source_ip_groups  = [azurerm_ip_group.workload_ip_group.id, azurerm_ip_group.infra_ip_group.id]
+    }
+  }
+  depends_on = [
```

Comment:
Already has implicit dependency in the `firewall_policy_id` property so can be removed.